packages/typescriptlang-org/src/components/search/joinSearchResults.ts

```diff
@@ -0,0 +1,39 @@
+import { JoinedPackage, JoinedSearchResult, RawSearchResults, SearchHit } from "./types"
+
+export const joinSearchResults = (
+    [directResult, typedResult]: RawSearchResults['results']
```

Comment:
So, here's a bit of maybe "wasted" work: the first few packages I scanned through didn't happen to mention `types: { definitelytyped: "@types/name" }`, so I figured we'd need to send _two_ search requests... but on looking deeper, it looks like Algolia actually does have the right types data in results... so, assuming that data is filled out nicely, we don't actually need this whole joining-two-searches-together shenanigan. 

Pushing this commit now as a reference in case it's useful later. 🤷